crates/stc_ts_types/src/lib.rs

```diff
@@ -1611,15 +1617,44 @@ impl Type {
         }
         tys.dedup_type();
 
```

Comment:
I think we should remove this logic instead of expanding

Reply:
all? or a bit?

I think we need things that look like the next code.
```rs
        if tys.iter().any(|ty| ty.is_never()) {
            return Type::never(span, Default::default());
        }
```

Reply:
Instead, I don't think I need the rest.

Reply:
I think we should rely on `normalize_intersection_types` instead of converting it to `never` on creation.

Reply:
but, I think that it need simple check for never.
That's how the original chord works.

Reply:
If so, can you extract this as `fn is_trivial_never(types: &[Type]) -> bool` of `Intersection`?

```rust
impl Intersection {
    pub fn fn is_trivial_never(types: &[Type]) -> bool {
    }
}
```

Reply:
Good. I think that's a good idea.
Is this the code?

Reply:
It should filled with logic in `new_intersection`

Reply:
What kind of logic?

I put in what I accidentally left out.
I broke the test because of them :D